lib/consumer_supervisor.ex

```diff
@@ -262,6 +258,23 @@ defmodule ConsumerSupervisor do
     call(supervisor, :count_children)
   end
 
+  @doc """
+  Receives a template to initialize and a set of options.
+
+  This is typically invoked at the end of the init/1 callback of module-based supervisors.
+
+  This function returns a the child specification and the supervisor flags.
+
+  """
+  def init([{_, _, _, _, _, _} = template], opts) do
+    {:ok, {_, spec}} = Supervisor.Spec.supervise([template], opts)
+    {:ok, spec, opts}
+  end
+  def init([template], opts) do
+    {:ok, {_, spec}} = Supervisor.init([template], opts)
```

Comment:
@josevalim 

I just saw that the tests failed for older Elixir Versions.

Before (Works with Elixir 1.4):

```elixir
  def init({_, _, _, _, _, _} = template, opts) do
    {:ok, {_, [template]}} = Supervisor.Spec.supervise([template], opts)
    {:ok, [template], opts}
  end
```